code/modules/mob/living/basic/snails/snail.dm

```diff
@@ -0,0 +1,113 @@
+	held_state = "snail"
+	basic_mob_flags = FLIP_ON_DEATH
+	head_icon = 'icons/mob/clothing/head/pets_head.dmi'
+	icon = 'icons/mob/simple/pets.dmi'
+	butcher_results = list(/obj/item/food/meat/rawcutlet = 1)
+	mob_biotypes = MOB_ORGANIC
+	health = 30
+	maxHealth = 30
+	speed = 6
+	verb_say = "gurgles"
+	verb_ask = "gurgles curiously"
+	can_be_held = TRUE
+	verb_exclaim = "gurgles loudly"
+	verb_yell = "gurgles loudly"
+	worn_slot_flags = ITEM_SLOT_HEAD
+	faction = list(FACTION_NEUTRAL, FACTION_MAINT_CREATURES)
+	ai_controller = /datum/ai_controller/basic_controller/snail
+
+/mob/living/basic/snail/Initialize(mapload)
+	. = ..()
+	var/static/list/loc_connections = list(COMSIG_ATOM_ENTERED = PROC_REF(on_entered))
+	AddElement(/datum/element/connect_loc, loc_connections)
+	var/static/list/eatable_food = list(
+		/obj/item/food/grown,
+		/obj/item/food/appleslice,
+	)
+
+	var/static/list/innate_actions = list(
+		/datum/action/cooldown/mob_cooldown/shell_retreat = BB_SNAIL_RETREAT_ABILITY,
+	)
+
+	grant_actions_by_list(innate_actions)
+	AddElement(/datum/element/ai_retaliate)
+	ai_controller.set_blackboard_key(BB_BASIC_FOODS, typecacheof(eatable_food))
+	AddElement(/datum/element/basic_eating, food_types = eatable_food)
+	create_reagents(100, REAGENT_HOLDER_ALIVE)
+	RegisterSignal(reagents, COMSIG_REAGENTS_HOLDER_UPDATED, PROC_REF(on_reagents_update))
+
+/mob/living/basic/snail/proc/on_entered(datum/source, obj/effect/decal/cleanable/food/salt/potential_salt)
+	SIGNAL_HANDLER
+	if(istype(potential_salt))
+		death() //immediately perish
+
+/mob/living/basic/snail/proc/on_reagents_update(datum/source)
+	SIGNAL_HANDLER
+	if(reagents.has_reagent(/datum/reagent/consumable/salt))
+		death()
+
+/mob/living/basic/snail/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if(!isturf(loc))
+		return
+	if(locate(/obj/effect/decal/cleanable/food/salt) in loc.contents)
+		death()
+
+/mob/living/basic/snail/mob_pickup(mob/living/user)
+	var/obj/item/clothing/head/mob_holder/snail/holder = new(get_turf(src), src, held_state, head_icon, held_lh, held_rh, worn_slot_flags)
+	var/display_message = "[user] [HAS_TRAIT(src, TRAIT_MOVE_FLOATING) ? "scoops up" : "peels [src] off the ground"]!"
+	user.visible_message(span_warning(display_message))
+	user.put_in_hands(holder)
+
+
+///snail's custom holder object
+/obj/item/clothing/head/mob_holder/snail
+
+/obj/item/clothing/head/mob_holder/snail/interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
+	if(!istype(interacting_with, /obj/machinery/hydroponics))
+		return NONE
+
+	. = ITEM_INTERACT_BLOCKING
+	if(held_mob.stat == DEAD)
+		user.balloon_alert(user, "appears to be dead!")
+		return
+
+	if(locate(type) in interacting_with)
+		user.balloon_alert(user, "already has a snail!")
+		return
+
+	if(!do_after(user, 2 SECONDS, interacting_with))
+		return
+
+	forceMove(interacting_with)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/effect/overlay/vis_effect/snail
```

Comment:
This might be better placed in the botany tray file or its own one, it's not really relevant to the mob particularly